Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import tensorflow as tf

def tanh(x):
    return 2. / (1. + tf.exp(-2 * x)) - 1.

x = tf.Variable(tf.linspace(-4, 4, 50))
with tf.GradientTape() as tape:
    tan_x = tanh(x)
grad_tanh_x = tape.gradient(tan_x, x)

plt.figure()
plt.plot(x.numpy(), tan_x.numpy(), label='tanh(x)')
plt.plot(x.numpy(), grad_tanh_x, label='tanh\'(x)')
plt.legend()
plt.grid('on');
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[2], line 1[0m
[0;32m----> 1[0m [38;5;28;01mimport[39;00m [38;5;21;01mtensorflow[39;00m [38;5;28;01mas[39;00m [38;5;21;01mtf[39;00m
[1;32m      3[0m [38;5;28;01mdef[39;00m [38;5;21mtanh[39m(x):
[1;32m      4[0m     [38;5;28;01mreturn[39;00m [38;5;241m2.[39m [38;5;241m/[39m ([38;5;241m1.[39m [38;5;241m+[39m tf[38;5;241m.[39mexp([38;5;241m-[39m[38;5;241m2[39m [38;5;241m*[39m x)) [38;5;241m-[39m [38;5;241m1.[39m

[0;31mModuleNotFoundError[0m: No module named 'tensorflow'

